java/flight/flight-core/src/main/java/org/apache/arrow/flight/CallOptions.java

```diff
@@ -56,7 +56,7 @@ public static CallOption timeout(long duration, TimeUnit unit) {
   /**
    * CallOptions specific to GRPC stubs.
    */
-  interface GrpcCallOption extends CallOption {
+  public interface GrpcCallOption extends CallOption {
```

Comment:
This can't be public in this module - it should have a public subinterface in flight-grpc.

Reply:
@lidavidm Moving `GrpcCallOption` to `flight-grpc` adds a circular dependency since `flight-grpc` is dependent on `flight-core` for FlightProducer and FlightClient.

Is the intention behind this comment to:
- move all grpc related dependencies to `flight-grpc` module?
- Or just move the `GrpcCallOption` to `org.apache.arrow.flight.grpc` package ?

Reply:
Hmm, that is a problem. We also don't want to expose gRPC details from the core package, though...leaving this as-is is probably simplest, though, seeing as a non-gRPC flight implementation is still a ways off.

Reply:
Thank you for the prompt response @lidavidm, keeping this as-is.